File: src_original/extras-kio/libauthinfo.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KIO__AuthInfo
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <QVariant>
#include <authinfo.h>
#include "libauthinfo.h"
#include "libauthinfo.hxx"

KIO__AuthInfo* KIO__AuthInfo_new() {
    return new KIO::AuthInfo();
}

KIO__AuthInfo* KIO__AuthInfo_new2(const KIO__AuthInfo* info) {
    return new KIO::AuthInfo(*info);
}

void KIO__AuthInfo_OperatorAssign(KIO__AuthInfo* self, const KIO__AuthInfo* info) {
    self->operator=(*info);
}

bool KIO__AuthInfo_IsModified(const KIO__AuthInfo* self) {
    return self->isModified();
}

void KIO__AuthInfo_SetModified(KIO__AuthInfo* self, bool flag) {
    self->setModified(flag);
}

QUrl* KIO__AuthInfo_Url(const KIO__AuthInfo* self) {
    return new QUrl(self->url);
}

void KIO__AuthInfo_SetUrl(KIO__AuthInfo* self, QUrl* url) {
    self->url = *url;
}

libqt_string KIO__AuthInfo_Username(const KIO__AuthInfo* self) {
    QString username_ret = self->username;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray username_b = username_ret.toUtf8();
    libqt_string username_str;
    username_str.len = username_b.length();
    username_str.data = static_cast<const char*>(malloc(username_str.len + 1));
    memcpy((void*)username_str.data, username_b.data(), username_str.len);
    ((char*)username_str.data)[username_str.len] = '\0';
    return username_str;
}

void KIO__AuthInfo_SetUsername(KIO__AuthInfo* self, libqt_string username) {
    QString username_QString = QString::fromUtf8(username.data, username.len);
    self->username = username_QString;
}

libqt_string KIO__AuthInfo_Password(const KIO__AuthInfo* self) {
    QString password_ret = self->password;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray password_b = password_ret.toUtf8();
    libqt_string password_str;
    password_str.len = password_b.length();
    password_str.data = static_cast<const char*>(malloc(password_str.len + 1));
    memcpy((void*)password_str.data, password_b.data(), password_str.len);
    ((char*)password_str.data)[password_str.len] = '\0';
    return password_str;
}

void KIO__AuthInfo_SetPassword(KIO__AuthInfo* self, libqt_string password) {
    QString password_QString = QString::fromUtf8(password.data, password.len);
    self->password = password_QString;
}

libqt_string KIO__AuthInfo_Prompt(const KIO__AuthInfo* self) {
    QString prompt_ret = self->prompt;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray prompt_b = prompt_ret.toUtf8();
    libqt_string prompt_str;
    prompt_str.len = prompt_b.length();
    prompt_str.data = static_cast<const char*>(malloc(prompt_str.len + 1));
    memcpy((void*)prompt_str.data, prompt_b.data(), prompt_str.len);
    ((char*)prompt_str.data)[prompt_str.len] = '\0';
    return prompt_str;
}

void KIO__AuthInfo_SetPrompt(KIO__AuthInfo* self, libqt_string prompt) {
    QString prompt_QString = QString::fromUtf8(prompt.data, prompt.len);
    self->prompt = prompt_QString;
}

libqt_string KIO__AuthInfo_Caption(const KIO__AuthInfo* self) {
    QString caption_ret = self->caption;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray caption_b = caption_ret.toUtf8();
    libqt_string caption_str;
    caption_str.len = caption_b.length();
    caption_str.data = static_cast<const char*>(malloc(caption_str.len + 1));
    memcpy((void*)caption_str.data, caption_b.data(), caption_str.len);
    ((char*)caption_str.data)[caption_str.len] = '\0';
    return caption_str;
}

void KIO__AuthInfo_SetCaption(KIO__AuthInfo* self, libqt_string caption) {
    QString caption_QString = QString::fromUtf8(caption.data, caption.len);
    self->caption = caption_QString;
}

libqt_string KIO__AuthInfo_Comment(const KIO__AuthInfo* self) {
    QString comment_ret = self->comment;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray comment_b = comment_ret.toUtf8();
    libqt_string comment_str;
    comment_str.len = comment_b.length();
    comment_str.data = static_cast<const char*>(malloc(comment_str.len + 1));
    memcpy((void*)comment_str.data, comment_b.data(), comment_str.len);
    ((char*)comment_str.data)[comment_str.len] = '\0';
    return comment_str;
}

void KIO__AuthInfo_SetComment(KIO__AuthInfo* self, libqt_string comment) {
    QString comment_QString = QString::fromUtf8(comment.data, comment.len);
    self->comment = comment_QString;
}

libqt_string KIO__AuthInfo_CommentLabel(const KIO__AuthInfo* self) {
    QString commentLabel_ret = self->commentLabel;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray commentLabel_b = commentLabel_ret.toUtf8();
    libqt_string commentLabel_str;
    commentLabel_str.len = commentLabel_b.length();
    commentLabel_str.data = static_cast<const char*>(malloc(commentLabel_str.len + 1));
    memcpy((void*)commentLabel_str.data, commentLabel_b.data(), commentLabel_str.len);
    ((char*)commentLabel_str.data)[commentLabel_str.len] = '\0';
    return commentLabel_str;
}

void KIO__AuthInfo_SetCommentLabel(KIO__AuthInfo* self, libqt_string commentLabel) {
    QString commentLabel_QString = QString::fromUtf8(commentLabel.data, commentLabel.len);
    self->commentLabel = commentLabel_QString;
}

libqt_string KIO__AuthInfo_RealmValue(const KIO__AuthInfo* self) {
    QString realmValue_ret = self->realmValue;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray realmValue_b = realmValue_ret.toUtf8();
    libqt_string realmValue_str;
    realmValue_str.len = realmValue_b.length();
    realmValue_str.data = static_cast<const char*>(malloc(realmValue_str.len + 1));
    memcpy((void*)realmValue_str.data, realmValue_b.data(), realmValue_str.len);
    ((char*)realmValue_str.data)[realmValue_str.len] = '\0';
    return realmValue_str;
}

void KIO__AuthInfo_SetRealmValue(KIO__AuthInfo* self, libqt_string realmValue) {
    QString realmValue_QString = QString::fromUtf8(realmValue.data, realmValue.len);
    self->realmValue = realmValue_QString;
}

libqt_string KIO__AuthInfo_DigestInfo(const KIO__AuthInfo* self) {
    QString digestInfo_ret = self->digestInfo;
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray digestInfo_b = digestInfo_ret.toUtf8();
    libqt_string digestInfo_str;
    digestInfo_str.len = digestInfo_b.length();
    digestInfo_str.data = static_cast<const char*>(malloc(digestInfo_str.len + 1));
    memcpy((void*)digestInfo_str.data, digestInfo_b.data(), digestInfo_str.len);
    ((char*)digestInfo_str.data)[digestInfo_str.len] = '\0';
    return digestInfo_str;
}

void KIO__AuthInfo_SetDigestInfo(KIO__AuthInfo* self, libqt_string digestInfo) {
    QString digestInfo_QString = QString::fromUtf8(digestInfo.data, digestInfo.len);
    self->digestInfo = digestInfo_QString;
}

bool KIO__AuthInfo_VerifyPath(const KIO__AuthInfo* self) {
    return self->verifyPath;
}

void KIO__AuthInfo_SetVerifyPath(KIO__AuthInfo* self, bool verifyPath) {
    self->verifyPath = verifyPath;
}

bool KIO__AuthInfo_ReadOnly(const KIO__AuthInfo* self) {
    return self->readOnly;
}

void KIO__AuthInfo_SetReadOnly(KIO__AuthInfo* self, bool readOnly) {
    self->readOnly = readOnly;
}

bool KIO__AuthInfo_KeepPassword(const KIO__AuthInfo* self) {
    return self->keepPassword;
}

void KIO__AuthInfo_SetKeepPassword(KIO__AuthInfo* self, bool keepPassword) {
    self->keepPassword = keepPassword;
}

void KIO__AuthInfo_SetExtraField(KIO__AuthInfo* self, const libqt_string fieldName, const QVariant* value) {
    QString fieldName_QString = QString::fromUtf8(fieldName.data, fieldName.len);
    self->setExtraField(fieldName_QString, *value);
}

void KIO__AuthInfo_SetExtraFieldFlags(KIO__AuthInfo* self, const libqt_string fieldName, const int flags) {
    QString fieldName_QString = QString::fromUtf8(fieldName.data, fieldName.len);
    self->setExtraFieldFlags(fieldName_QString, static_cast<const KIO::AuthInfo::FieldFlags>(flags));
}

QVariant* KIO__AuthInfo_GetExtraField(const KIO__AuthInfo* self, const libqt_string fieldName) {
    QString fieldName_QString = QString::fromUtf8(fieldName.data, fieldName.len);
    return new QVariant(self->getExtraField(fieldName_QString));
}

int KIO__AuthInfo_GetExtraFieldFlags(const KIO__AuthInfo* self, const libqt_string fieldName) {
    QString fieldName_QString = QString::fromUtf8(fieldName.data, fieldName.len);
    return static_cast<int>(self->getExtraFieldFlags(fieldName_QString));
}

void KIO__AuthInfo_RegisterMetaTypes() {
    KIO::AuthInfo::registerMetaTypes();
}

void KIO__AuthInfo_Delete(KIO__AuthInfo* self) {
    delete self;
}

File: include_original/extras-kio/libauthinfo.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const authinfo_enums = enums;
const std = @import("std");

/// https://api.kde.org/kio-authinfo.html
pub const kio__authinfo = struct {
    /// New constructs a new KIO::AuthInfo object.
    ///
    ///
    pub fn New() QtC.KIO__AuthInfo {
        return qtc.KIO__AuthInfo_new();
    }

    /// New2 constructs a new KIO::AuthInfo object.
    ///
    /// ``` info: QtC.KIO__AuthInfo ```
    pub fn New2(info: ?*anyopaque) QtC.KIO__AuthInfo {
        return qtc.KIO__AuthInfo_new2(@ptrCast(info));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#operator-eq)
    ///
    /// ``` self: QtC.KIO__AuthInfo, info: QtC.KIO__AuthInfo ```
    pub fn OperatorAssign(self: ?*anyopaque, info: ?*anyopaque) void {
        qtc.KIO__AuthInfo_OperatorAssign(@ptrCast(self), @ptrCast(info));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#isModified)
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn IsModified(self: ?*anyopaque) bool {
        return qtc.KIO__AuthInfo_IsModified(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#setModified)
    ///
    /// ``` self: QtC.KIO__AuthInfo, flag: bool ```
    pub fn SetModified(self: ?*anyopaque, flag: bool) void {
        qtc.KIO__AuthInfo_SetModified(@ptrCast(self), flag);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#url-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn Url(self: ?*anyopaque) QtC.QUrl {
        return qtc.KIO__AuthInfo_Url(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#url-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, url: QtC.QUrl ```
    pub fn SetUrl(self: ?*anyopaque, url: QtC.QUrl) void {
        qtc.KIO__AuthInfo_SetUrl(@ptrCast(self), @ptrCast(url));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#username-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn Username(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const username_str = qtc.KIO__AuthInfo_Username(@ptrCast(self));
        defer qtc.libqt_string_free(&username_str);
        const username_ret = allocator.alloc(u8, username_str.len) catch @panic("kio::authinfo.Username: Memory allocation failed");
        @memcpy(username_ret, username_str.data[0..username_str.len]);
        return username_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#username-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, username: []const u8 ```
    pub fn SetUsername(self: ?*anyopaque, username: []const u8) void {
        const username_str = qtc.libqt_string{
            .len = username.len,
            .data = username.ptr,
        };
        qtc.KIO__AuthInfo_SetUsername(@ptrCast(self), username_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#password-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn Password(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const password_str = qtc.KIO__AuthInfo_Password(@ptrCast(self));
        defer qtc.libqt_string_free(&password_str);
        const password_ret = allocator.alloc(u8, password_str.len) catch @panic("kio::authinfo.Password: Memory allocation failed");
        @memcpy(password_ret, password_str.data[0..password_str.len]);
        return password_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#password-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, password: []const u8 ```
    pub fn SetPassword(self: ?*anyopaque, password: []const u8) void {
        const password_str = qtc.libqt_string{
            .len = password.len,
            .data = password.ptr,
        };
        qtc.KIO__AuthInfo_SetPassword(@ptrCast(self), password_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#prompt-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn Prompt(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const prompt_str = qtc.KIO__AuthInfo_Prompt(@ptrCast(self));
        defer qtc.libqt_string_free(&prompt_str);
        const prompt_ret = allocator.alloc(u8, prompt_str.len) catch @panic("kio::authinfo.Prompt: Memory allocation failed");
        @memcpy(prompt_ret, prompt_str.data[0..prompt_str.len]);
        return prompt_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#prompt-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, prompt: []const u8 ```
    pub fn SetPrompt(self: ?*anyopaque, prompt: []const u8) void {
        const prompt_str = qtc.libqt_string{
            .len = prompt.len,
            .data = prompt.ptr,
        };
        qtc.KIO__AuthInfo_SetPrompt(@ptrCast(self), prompt_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#caption-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn Caption(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const caption_str = qtc.KIO__AuthInfo_Caption(@ptrCast(self));
        defer qtc.libqt_string_free(&caption_str);
        const caption_ret = allocator.alloc(u8, caption_str.len) catch @panic("kio::authinfo.Caption: Memory allocation failed");
        @memcpy(caption_ret, caption_str.data[0..caption_str.len]);
        return caption_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#caption-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, caption: []const u8 ```
    pub fn SetCaption(self: ?*anyopaque, caption: []const u8) void {
        const caption_str = qtc.libqt_string{
            .len = caption.len,
            .data = caption.ptr,
        };
        qtc.KIO__AuthInfo_SetCaption(@ptrCast(self), caption_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#comment-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn Comment(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const comment_str = qtc.KIO__AuthInfo_Comment(@ptrCast(self));
        defer qtc.libqt_string_free(&comment_str);
        const comment_ret = allocator.alloc(u8, comment_str.len) catch @panic("kio::authinfo.Comment: Memory allocation failed");
        @memcpy(comment_ret, comment_str.data[0..comment_str.len]);
        return comment_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#comment-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, comment: []const u8 ```
    pub fn SetComment(self: ?*anyopaque, comment: []const u8) void {
        const comment_str = qtc.libqt_string{
            .len = comment.len,
            .data = comment.ptr,
        };
        qtc.KIO__AuthInfo_SetComment(@ptrCast(self), comment_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#commentLabel-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn CommentLabel(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const commentLabel_str = qtc.KIO__AuthInfo_CommentLabel(@ptrCast(self));
        defer qtc.libqt_string_free(&commentLabel_str);
        const commentLabel_ret = allocator.alloc(u8, commentLabel_str.len) catch @panic("kio::authinfo.CommentLabel: Memory allocation failed");
        @memcpy(commentLabel_ret, commentLabel_str.data[0..commentLabel_str.len]);
        return commentLabel_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#commentLabel-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, commentLabel: []const u8 ```
    pub fn SetCommentLabel(self: ?*anyopaque, commentLabel: []const u8) void {
        const commentLabel_str = qtc.libqt_string{
            .len = commentLabel.len,
            .data = commentLabel.ptr,
        };
        qtc.KIO__AuthInfo_SetCommentLabel(@ptrCast(self), commentLabel_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#realmValue-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn RealmValue(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const realmValue_str = qtc.KIO__AuthInfo_RealmValue(@ptrCast(self));
        defer qtc.libqt_string_free(&realmValue_str);
        const realmValue_ret = allocator.alloc(u8, realmValue_str.len) catch @panic("kio::authinfo.RealmValue: Memory allocation failed");
        @memcpy(realmValue_ret, realmValue_str.data[0..realmValue_str.len]);
        return realmValue_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#realmValue-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, realmValue: []const u8 ```
    pub fn SetRealmValue(self: ?*anyopaque, realmValue: []const u8) void {
        const realmValue_str = qtc.libqt_string{
            .len = realmValue.len,
            .data = realmValue.ptr,
        };
        qtc.KIO__AuthInfo_SetRealmValue(@ptrCast(self), realmValue_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#digestInfo-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, allocator: std.mem.Allocator ```
    pub fn DigestInfo(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const digestInfo_str = qtc.KIO__AuthInfo_DigestInfo(@ptrCast(self));
        defer qtc.libqt_string_free(&digestInfo_str);
        const digestInfo_ret = allocator.alloc(u8, digestInfo_str.len) catch @panic("kio::authinfo.DigestInfo: Memory allocation failed");
        @memcpy(digestInfo_ret, digestInfo_str.data[0..digestInfo_str.len]);
        return digestInfo_ret;
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#digestInfo-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, digestInfo: []const u8 ```
    pub fn SetDigestInfo(self: ?*anyopaque, digestInfo: []const u8) void {
        const digestInfo_str = qtc.libqt_string{
            .len = digestInfo.len,
            .data = digestInfo.ptr,
        };
        qtc.KIO__AuthInfo_SetDigestInfo(@ptrCast(self), digestInfo_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#verifyPath-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn VerifyPath(self: ?*anyopaque) bool {
        return qtc.KIO__AuthInfo_VerifyPath(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#verifyPath-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, verifyPath: bool ```
    pub fn SetVerifyPath(self: ?*anyopaque, verifyPath: bool) void {
        qtc.KIO__AuthInfo_SetVerifyPath(@ptrCast(self), verifyPath);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#readOnly-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn ReadOnly(self: ?*anyopaque) bool {
        return qtc.KIO__AuthInfo_ReadOnly(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#readOnly-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, readOnly: bool ```
    pub fn SetReadOnly(self: ?*anyopaque, readOnly: bool) void {
        qtc.KIO__AuthInfo_SetReadOnly(@ptrCast(self), readOnly);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#keepPassword-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn KeepPassword(self: ?*anyopaque) bool {
        return qtc.KIO__AuthInfo_KeepPassword(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#keepPassword-var)
    ///
    /// ``` self: QtC.KIO__AuthInfo, keepPassword: bool ```
    pub fn SetKeepPassword(self: ?*anyopaque, keepPassword: bool) void {
        qtc.KIO__AuthInfo_SetKeepPassword(@ptrCast(self), keepPassword);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#setExtraField)
    ///
    /// ``` self: QtC.KIO__AuthInfo, fieldName: []const u8, value: QtC.QVariant ```
    pub fn SetExtraField(self: ?*anyopaque, fieldName: []const u8, value: ?*anyopaque) void {
        const fieldName_str = qtc.libqt_string{
            .len = fieldName.len,
            .data = fieldName.ptr,
        };
        qtc.KIO__AuthInfo_SetExtraField(@ptrCast(self), fieldName_str, @ptrCast(value));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#setExtraFieldFlags)
    ///
    /// ``` self: QtC.KIO__AuthInfo, fieldName: []const u8, flags: authinfo_enums.FieldFlags ```
    pub fn SetExtraFieldFlags(self: ?*anyopaque, fieldName: []const u8, flags: i32) void {
        const fieldName_str = qtc.libqt_string{
            .len = fieldName.len,
            .data = fieldName.ptr,
        };
        qtc.KIO__AuthInfo_SetExtraFieldFlags(@ptrCast(self), fieldName_str, @intCast(flags));
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#getExtraField)
    ///
    /// ``` self: QtC.KIO__AuthInfo, fieldName: []const u8 ```
    pub fn GetExtraField(self: ?*anyopaque, fieldName: []const u8) QtC.QVariant {
        const fieldName_str = qtc.libqt_string{
            .len = fieldName.len,
            .data = fieldName.ptr,
        };
        return qtc.KIO__AuthInfo_GetExtraField(@ptrCast(self), fieldName_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#getExtraFieldFlags)
    ///
    /// ``` self: QtC.KIO__AuthInfo, fieldName: []const u8 ```
    ///
    /// Returns: ``` authinfo_enums.FieldFlags ```
    pub fn GetExtraFieldFlags(self: ?*anyopaque, fieldName: []const u8) i32 {
        const fieldName_str = qtc.libqt_string{
            .len = fieldName.len,
            .data = fieldName.ptr,
        };
        return qtc.KIO__AuthInfo_GetExtraFieldFlags(@ptrCast(self), fieldName_str);
    }

    /// [Qt documentation](https://api.kde.org/kio-authinfo.html#registerMetaTypes)
    ///
    ///
    pub fn RegisterMetaTypes() void {
        qtc.KIO__AuthInfo_RegisterMetaTypes();
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KIO__AuthInfo ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KIO__AuthInfo_Delete(@ptrCast(self));
    }
};

/// https://api.kde.org/kio-authinfo.html#types
pub const enums = struct {
    pub const FieldFlags = enum {
        pub const ExtraFieldNoFlags: i32 = 0;
        pub const ExtraFieldReadOnly: i32 = 2;
        pub const ExtraFieldMandatory: i32 = 4;
    };
};
